### PR TITLE
Added logging to astronomy pipeline.

### LINKED ISSUE #138: Add astronomy pipeline logging
As a developer or user, when my code runs I want frequent logs describing information and errors so that I can transparently see how the code is running in both good and bad functionality.

### PR DESCRIPTION
**Pull request by olliejth**
Added basic logging to the astronomy pipeline at the INFO level.

closes #138

### CODE REVIEW

File: astronomy-pipeline/astronomy_extract.py [hunk 1]
```diff
@@ -196,7 +196,9 @@ def save_to_file(filename: str, data: list[dict]) -> None:
 def extract_weekly_astronomy_data():
     """Main function for extracting astronomical for a week"""
 
-    start_date = date.today()  # + timedelta(days=7)
+    logging.info("Data extraction started.")
+
+    start_date = date.today() + timedelta(days=7)
```
Comment (AlexOBarnes) [inline astronomy-pipeline/astronomy_extract.py:202]: This is very important for making sure tasks are executed

File: astronomy-pipeline/astronomy_load.py [hunk 2]
```diff
@@ -44,10 +49,14 @@ def upload_moon_phase_data(moon_phase_data: list[list]) -> None:
                 %s"""
 
         execute_values(cur, q_str, moon_phase_data)
+        logging.info("Moon phase data uploaded.")
```
Comment (AlexOBarnes) [inline astronomy-pipeline/astronomy_load.py:52]: Informative

File: astronomy-pipeline/astronomy_transform.py [hunk 3]
```diff
@@ -134,15 +135,19 @@ def convert_moon_datetime(moon_list: list) -> list:
 def transform_astronomy_data(raw_data: dict) -> list:
     """Main function for converting the extracted astronomy data into flat dataframes."""
 
+    logging.info("Data cleaning started.")
+
     merged_df = get_data_into_dataframe(raw_data)
 
     position_list = clean_position_data(merged_df)
     position_list = convert_positions_datetime(position_list)
+    logging.info("Body position data converted to 2D list.")
```
Comment (AlexOBarnes) [inline astronomy-pipeline/astronomy_transform.py:144]: Very informative

File: astronomy-pipeline/test_astronomy_extract.py [hunk 4]
```diff
@@ -124,6 +124,7 @@ class MockResponse():
             def __init__(self, data: dict):
                 """Init functions for mock class."""
                 self._data = data
+                self.status_code = 200
```
Comment (AlexOBarnes) [inline astronomy-pipeline/test_astronomy_extract.py:127]: Interesting use of mock class

File: astronomy-pipeline/astronomy_extract.py [hunk 5]
```diff
@@ -7,12 +7,12 @@
 from datetime import date, timedelta, datetime
 import base64
 import json
+import logging
```
Comment (CalebMathers) [inline astronomy-pipeline/astronomy_extract.py:10]: Definitely a good thing to import if you're adding logging

File: astronomy-pipeline/astronomy_extract.py [hunk 6]
```diff
@@ -196,7 +196,9 @@ def save_to_file(filename: str, data: list[dict]) -> None:
 def extract_weekly_astronomy_data():
     """Main function for extracting astronomical for a week"""
 
-    start_date = date.today()  # + timedelta(days=7)
+    logging.info("Data extraction started.")
```
Comment (CalebMathers) [inline astronomy-pipeline/astronomy_extract.py:199]: Clear, concise, and compelling

File: astronomy-pipeline/astronomy_extract.py [hunk 7]
```diff
@@ -208,10 +210,12 @@ def extract_weekly_astronomy_data():
 
     position_data = get_position_data(
         output_dict, times, regions, start_date, end_date)
+    logging.info("Body position data extracted and refined.")
 
     final_dict = {}
     final_dict["body_positions"] = position_data
     final_dict["moon_phase_urls"] = get_moon_urls(start_date)
+    logging.info("Moon phase data extracted.")
```
Comment (CalebMathers) [inline astronomy-pipeline/astronomy_extract.py:218]: Very informative

File: astronomy-pipeline/astronomy_pipeline.py [hunk 8]
```diff
@@ -2,24 +2,38 @@
...
+import logging
+
 from astronomy_extract import extract_weekly_astronomy_data
 from astronomy_transform import transform_astronomy_data
 from astronomy_load import upload_astronomy_data
 
 
 def lambda_handler(event=None, context=None) -> None:
     '''Runs the notification pipeline'''
-    print('Running Lambda')
```
Comment (CalebMathers) [inline astronomy-pipeline/astronomy_pipeline.py:12]: Nice to get this gone ready for prod

File: astronomy-pipeline/astronomy_pipeline.py [hunk 9]
```diff
@@ -2,24 +2,38 @@
...
     # {
     #   "positions_list": body_position_list,
     #   "moon_phase_list": moon_phase_list
     # }
     transformed_data = transform_astronomy_data(extract_data)
+    logging.info("Astronomy data transformation complete.")
 
     upload_astronomy_data(transformed_data)
+    logging.info("Astronomy data upload complete.")
+    logging.info("Astronomy execution time: %s seconds" %
```
Comment (CalebMathers) [inline astronomy-pipeline/astronomy_pipeline.py:35]: Love this

**Review by CalebMathers — APPROVED** [review]

**Pull request merged**